index.js

```diff
@@ -0,0 +1,179 @@
+const fs = require('fs');
+
+const rainbowLib = require('rainbow-bridge-lib');
+const { BorshContract } = require('rainbow-bridge-lib/rainbow/borsh');
+const {
+    EthProofExtractor,
+    receiptFromWeb3,
+    logFromWeb3,
+} = require('rainbow-bridge-lib/eth-proof-extractor');
+const { rlp } = require('ethereumjs-util')
+
+const utils = rainbowLib.utils;
+const nearAPI = rainbowLib.nearAPI;
+
+const NEAR_CONNECTOR_DEPOSIT = nearAPI.utils.format.parseNearAmount('50');
+const NEAR_CONNECTOR_NEW_GAS = '100000000000000';
+const ETH_CONNECTOR_DEPLOY_GAS = 5000000;
+const ETH_BRIDGE_TOKEN_DEPLOY_GAS = 5000000;
+const NEAR_BRIDGE_TOKEN_DEPOSIT = nearAPI.utils.format.parseNearAmount('40');
+const NEAR_BRIDGE_TOKEN_DEPLOY_GAS = '100000000000000';
+
+const NEAR_CONNECTOR_SCHEMA = {
+    bool: {
+        kind: 'function',
+        // @ts-ignore
+        ser: (b) => Buffer.from(Web3.utils.hexToBytes(b ? '0x01' : '0x00')),
+        deser: (z) => readerToHex(1)(z) === '0x01',
+    },
+    Proof: {
+        kind: 'struct',
+        fields: [
+            ['log_index', 'u64'],
+            ['log_entry_data', ['u8']],
+            ['receipt_index', 'u64'],
+            ['receipt_data', ['u8']],
+            ['header_data', ['u8']],
+            ['proof', [['u8']]],
+        ],
+    },
+}
+
+async function createNearConnector(masterAccount, config) {
+    if (!await rainbowLib.utils.accountExists(masterAccount.connection, config.nearConnectorId)) {
+        const code = fs.readFileSync('res/bridge_token_factory.wasm');
+        const newArgs = {
+            prover_account: config.nearEthProverId,
+            locker_address: utils.remove0x(config.ethConnectorAddress),
+        };
+        await masterAccount.signAndSendTransaction(config.nearConnectorId, [
+            nearAPI.transactions.createAccount(),
+            nearAPI.transactions.transfer(NEAR_CONNECTOR_DEPOSIT),
+            nearAPI.transactions.deployContract(code),
+            nearAPI.transactions.functionCall('new', newArgs, NEAR_CONNECTOR_NEW_GAS)
+        ]);
+    }
+    return new BorshContract(NEAR_CONNECTOR_SCHEMA, masterAccount, config.nearConnectorId, {
+        viewMethods: ['get_bridge_token_account_id'],
+        changeMethods: [
+            { methodName: 'deposit', inputFieldType: 'Proof', outputFieldType: null },
+            // {'deploy_bridge_token'}, 
+            // {'lock'}, 
+            // {'unlock'}],
+        ]
+    });
+}
+
+async function createEthConnector(web3, config) {
```

Comment:
I'd rather see destructuring used here instead of passing an opaque `config` so that it's more obvious what values need to be passed in

TypeScript would be even better! 😄